src/chainbin/chain-make-num-fst-e2e.cc

```diff
@@ -0,0 +1,121 @@
+// chainbin/chain-make-num-fst-e2e.cc
+
+// Copyright 2019  Johns Hopkins University (author: Yiwen Shao)
+
+// See ../../COPYING for clarification regarding multiple authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY IMPLIED
+// WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABLITY OR NON-INFRINGEMENT.
+// See the Apache 2 License for the specific language governing permissions and
+// limitations under the License.
+
+/** @brief Converts fsts (containing transition-ids) to fsts (containing pdf-ids, zero-based).
+*/
+#include "base/kaldi-common.h"
+#include "gmm/am-diag-gmm.h"
+#include "hmm/transition-model.h"
+#include "hmm/hmm-utils.h"
+#include "util/common-utils.h"
+#include "fst/fstlib.h"
+
+namespace kaldi {
+
+bool FstTransitionToPdf(const fst::StdVectorFst &fst_transition,
```

Comment:
I think it should be .. ToPdfPlusOne(), since these are pdf+1.  Also I think this conflicts with your comment or commit message in which you say they are zero-based, these seem to be one-based.